pkg/gds/certman/certs_test.go

```diff
@@ -205,6 +205,166 @@ func (s *certTestSuite) TestCertManager() {
+	require.NoError(err, "could not get echo VASP")
+	echoVASP = s.setupVASP(echoVASP)
+
+	// Call the certman function at 29 days, which will send
+	// the thirty day cert reissuance reminder to echoVASP and
+	// the TRISA admin.
+	s.updateVaspIdentityCert(echoVASP, 29)
+	callTime := time.Now()
+	s.certman.HandleCertificateReissuance()
+
+	v, err := s.db.RetrieveVASP(echoVASP.Id)
+	require.NoError(err)
+
+	// Ensure that the expected emails have been sent, using
+	// the mock email client.
+	messages := []*emails.EmailMeta{
+		{
+			To:        v.Contacts.Technical.Email,
+			From:      s.conf.Email.ServiceEmail,
+			Subject:   emails.ReissuanceReminderRE,
+			Reason:    "reissuance_reminder",
+			Timestamp: callTime,
+		},
+		{
+			To:        s.conf.Email.AdminEmail,
+			From:      s.conf.Email.ServiceEmail,
+			Subject:   emails.ExpiresAdminNotificationRE,
+			Reason:    "expires_admin_notification",
+			Timestamp: callTime,
+		},
+	}
+	emails.CheckEmails(s.T(), messages)
+}
+
+func (s *certTestSuite) TestCertManagerSevenDayReissuanceReminder() {
+	_ = s.setupCertManager(sectigo.ProfileCipherTraceEE)
+	defer s.teardownCertManager()
+	defer s.fixtures.LoadReferenceFixtures()
+	require := s.Require()
+
+	// setup the datastore to contain the modified echoVASP
+	s.resetVASPDatabase()
+	echoVASP, err := s.fixtures.GetVASP("echo")
+	require.NoError(err, "could not get echo VASP")
+	echoVASP = s.setupVASP(echoVASP)
+
+	// Call the certman function at 6 days, which will send
+	// the seven day cert reissuance reminder to echoVASP.
+	s.updateVaspIdentityCert(echoVASP, 6)
+	callTime := time.Now()
+	s.certman.HandleCertificateReissuance()
+
+	v, err := s.db.RetrieveVASP(echoVASP.Id)
+	require.NoError(err)
+
+	// Ensure that the expected email has been sent, using
+	// the mock email client.
+	messages := []*emails.EmailMeta{
+		{
+			To:        v.Contacts.Technical.Email,
+			From:      s.conf.Email.ServiceEmail,
+			Subject:   emails.ReissuanceReminderRE,
+			Reason:    "reissuance_reminder",
+			Timestamp: callTime,
+		},
+	}
+	emails.CheckEmails(s.T(), messages)
+}
+
+func (s *certTestSuite) TestCertManagerReissuance() {
+	_ = s.setupCertManager(sectigo.ProfileCipherTraceEE)
+	defer s.teardownCertManager()
+	defer s.fixtures.LoadReferenceFixtures()
+
+	// setup the datastore to contain the modified echoVASP
+	require := s.Require()
+	s.resetVASPDatabase()
+	echoVASP, err := s.fixtures.GetVASP("echo")
+	require.NoError(err, "could not get echo VASP")
+	echoVASP = s.setupVASP(echoVASP)
+
+	// Call the certman function at 8 days, which should
+	// reissue the VASP's identity certificate, send the
+	// email with the created pkcs12 password and send
+	// the whisper link, as well as notifying the TRISA
+	// admin that reissuance has started.
+	s.updateVaspIdentityCert(echoVASP, 8)
+	callTime := time.Now()
+	s.certman.HandleCertificateReissuance()
+
+	v, err := s.db.RetrieveVASP(echoVASP.Id)
+	require.NoError(err)
```

Comment:
Since this starts the reissuance process, I'd also like to add a few more asserts here:

1. The certificate request is created and the state is in `READY_TO_SUBMIT`
2. A secret has been created in the secret manager
3. After two calls of `HandleCertificateRequests`, the VASP contains a new identity certificate, the cert request is completed, and there is a new certificate record in the database. (this might be too complicated for this PR so I'm fine with creating a follow up story for this)

Let me know if you need any help with the first two points.